hpx/lcos/detail/future_data.hpp

```diff
@@ -357,7 +348,7 @@ namespace detail
         {
             result_type* value_ptr = reinterpret_cast<result_type*>(&storage_);
             construct(value_ptr);
-            state_ = value;
+            state_.store(value, std::memory_order_relaxed);
```

Comment:
the next constructor should probably use relaxed too.

Reply:
It already does.

Reply:
no, there's another constructor in between. github doesn't let me comment on it since it's not part of the diff